homeassistant/components/aws/image_processing.py

```diff
@@ -0,0 +1,483 @@
+        # Update entity store
+        self._objects = objects
+        self._labels = labels
+
+    async def compute_objects(self, client, image):
+        """Detect labels in image and parse response."""
+        detect_labels = await client.detect_labels(Image={"Bytes": image})
+        objects, labels = await get_objects(detect_labels, self.confidence)
+        return objects, labels
+
+    async def async_process_image(self, image):
+        """Process image.
+
+        This method is a coroutine.
+        """
+        async with self.session.create_client(
+            self._service, **self.session_config
+        ) as client:
+            objects, labels = await self.compute_objects(client, image)
+            if len(objects) > 0 and self.conf_save_file_folder:
+                await save_image(
+                    self,
+                    image,
+                    self.name,
+                    objects,
+                    self.conf_save_file_folder,
+                    self.conf_save_file_timestamp,
+                )
+            await self.async_process_objects(objects, labels)
+
+
+class RekognitionFaceEntity(ImageProcessingFaceEntity):
+    """Rekognition Face Detection/Identification capabilities."""
+
+    _service = "rekognition"
+
+    def __init__(
+        self, camera_entity, session, session_config, entity_config, name=None,
+    ):
+        """Initialize the Rekognition service."""
+        super().__init__()
+        self.session = session
+        self.session_config = session_config
+        self.camera = camera_entity
+        self.collection_id = entity_config[CONF_COLLECTION_ID]
+        self.identify_faces = entity_config.get(CONF_IDENTIFY_FACES, False)
+        self.conf_save_file_folder = entity_config.get(CONF_SAVE_FILE_FOLDER, None)
+        self.conf_save_file_timestamp = entity_config.get(
+            CONF_SAVE_FILE_TIMESTAMP, False
+        )
+        self._confidence = entity_config.get(CONF_CONFIDENCE, 0.0)
+        self.detection_attributes = entity_config.get(
+            CONF_DETECTION_ATTRIBUTES, "DEFAULT"
+        )
+        self.valid_collection_id = False
+        if name:
+            self._name = name
+        else:
+            self._name = f"Rekognition Face {split_entity_id(camera_entity)[1]}"
+
+    @property
+    def confidence(self):
+        """Return minimum confidence for send events."""
+        return self._confidence
+
+    @property
+    def camera_entity(self):
+        """Return camera entity id from process pictures."""
+        return self.camera
+
+    @property
+    def name(self):
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def should_poll(self):
+        """Return the polling state."""
+        return False
+
+    async def init_collection(self, client):
+        """Create AWS image collection if doesn't exist."""
+        collection_id_req = await client.list_collections(MaxResults=99)
```

Comment:
What does `MaxResults` impact?

Reply:
Nothing - I thought I saw `list_collections` had a low default result limit to return but I can't seem to find that now. Should be safe to remove and it will return the full list.